File: ch9/streaming/wc/python/reduce.py
#!/bin/env python
import sys

count = 1
current = None

for word in sys.stdin:
    word = word.strip()

    if word == current:
        count += 1
    else:
        if current:
            print "%s\t%s" % (current.decode('utf-8'), count)
        current = word
        count = 1
if current == word:
    print "%s\t%s" % (current.decode('utf-8'), count)

File: ch9/streaming/wc/ruby/reduce.rb
#!/usr/bin/env ruby         

current = nil
 count = 0
 while line = gets
word, counter = line.split("\t")
if word == current
count = count+1
else
puts current+"\t"+count.to_s if current
current = word
count = 1
 end end         
puts current+"\t"+count.to_s

